### ISSUE TITLE
when clause support check file is symbolic link

### ISSUE THREAD

**Issue by tjx666** [microsoft/vscode#167996]
I want to make an extension to provide context menu which named `go to origin file` in file explorer when selected file/folder is symbolic link.

**Comment by lramos15** [comment]
No plans to address this in the near future. You may be able to utilize the workspace root from the API to scan the file system and resolve what is an isn't a symbolic link locally to create your own context key in your extension.

**Comment by tjx666** [comment]
I have two questions.

1. If I need to create the context key, I can only set the extension active event to `*` or `onStartupFinished`.
2. There seems no api to listen the explorer context menu event

Can you provide more details info about how to implement it?

My extension code is here: https://github.com/tjx666/vscode-neo-file-utils/blob/main/src/extension.ts

**Comment by lramos15** [comment]
> 1. If I need to create the context key, I can only set the extension active event to `*` or `onStartupFinished`.

Yes, this isn't ideal but neither is maintaining your own context key it's just at the moment it is not planned for the nearby future and we try to prevent collecting a ton of issues that just end up sitting.



> 2. There seems no api to listen the explorer context menu event

Why would you need this? You would just put the when clause in the menu registration

**Comment by tjx666** [comment]
What I want implement:

When user open the explorer file menu, the menu item will show `Open Symbolic Link Real File` menu item only if the file is a aymbolic link.

What problem when I using context key?

If I understand right, I need to create the context key `neo-file-utils.isSymbolicLink` before user open the explorer menu. So, the question is how can I register the context key before user open explorer menu.

@lramos15
